src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +122,49 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter);
         }
     };
 }
 
 using namespace RSS;
 
+QString computeEpisodeName(const QString &article)
+{
+    static const QRegularExpression episodeRegex(
+                    "(?:^|[^A-Za-z0-9])(?:"
+
+                    //Format 1: s01e01
+                    "(?:s(\\d{1,})e(\\d{1,}))|"
+
+                    //Format 2: 01x01
+                    "(?:(\\d{1,})x(\\d{1,}))|"
+
+                    //Format 3: 2017.01.01
+                    "(\\d{4}\\.\\d{2}\\.\\d{2})"
+
+                    ")(?:[^A-Za-z0-9]|$)",
```

Comment:
@elFarto I don't think that regex is too lenient. It's really unlikely to match anything other than date, especially since either the first or last group of numbers has to be four digits long. If you wanted you could modify it to exclude `_` and perhaps `x` as a separator character. I think it's likely that there will be dates separated with hyphens since that's standard in a lot of parts of the world.

As for reformatting it... the Correct™ solution probably **is** to reformat the date into a canonical format. Eventually there will be two torrents with the same date in two different formats for someone. But that's a pretty unlikely edge case, and on top of that, there's cases where it's ambiguous what the date actually is. (e.g. does 01.02.2017 mean January 2 or February 1?) So I don't think this is actually necessary.